### PR TITLE
docs: reorganize reference docs and sidebar

### PR DESCRIPTION
**Pull request by jpetey75**
Closes: https://github.com/lightdash/lightdash/issues/11178

### Description:

Reorganize sidebar (see linked issue) and move some docs from guides to references.

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: docs/sidebars.js [hunk 1]
```diff
@@ -126,63 +113,174 @@ module.exports = {
...
+                        },
+                        {
+                            type: 'doc',
+                            id: 'guides/cli/how-to-auto-generate-schema-files',
+                            label: 'Lightdash Generate',
+                        },
+                        {
+                            type: 'doc',
+                            id: 'guides/cli/how-to-use-lightdash-preview',
+                            label: 'Lightdash Generate',
```
Comment (TuringLovesDeathMetal) [inline docs/sidebars.js:163]:
```suggestion
                            label: 'Lightdash Preview',
```

File: docs/sidebars.js [hunk 2]
```diff
@@ -126,63 +113,174 @@ module.exports = {
...
+                            label: 'Lightdash Validate',
+                        },        
+                    ]
                 },
                 {
-                    type: 'doc',
-                    id: 'references/metrics',
-                    label: 'Metrics',
+                    type: 'category',
+                    label: 'Analytics Engineering',
```
Comment (TuringLovesDeathMetal) [inline docs/sidebars.js:184]: I'm not sure about this label...like, I get the sentiment, but I thought that analytics engineering best principles was going to be under this reference doc. But actually, it's all about the semantic layer (is that analytics engineering?)

Maybe we could call it something like `developing your project` or `Lightdash development`?

**Review by TuringLovesDeathMetal — CHANGES_REQUESTED** [review]
Nice! Looks great - just two small comments from me.

File: docs/sidebars.js (hunk 2 above)
Comment (jpetey75) [reply docs/sidebars.js:184]: @TuringLovesDeathMetal  I agree and I also like `Lightdash Development` better so i'll make that update

**Comment by jpetey75** [comment]
@TuringLovesDeathMetal can you re-approve if changes all look good?

**Review by TuringLovesDeathMetal — APPROVED** [review]
looks great! well done on this 💪

**Pull request merged**